### PR TITLE
Please consider the minor fixes

### PR DESCRIPTION
**Pull request by tkrisnguyen**
IMHO, the fixes are already self-explainable.

### CODE REVIEW

**Comment by jorgepz** [comment]
Thanks @tkrisnguyen.  @mvanzulli or @mforets . do you have any comments on this pr?

File: src/Entities/Entities.jl [hunk 1]
```diff
@@ -10,7 +10,7 @@ module Entities
 
 using Reexport, Dictionaries, StaticArrays
 
-using ..Nodes
+# using ..Nodes
```
Comment (mvanzulli) [inline src/Entities/Entities.jl]: Minor: Seems that this dependency is needed . So I would suggest to revert the change 
```
Failed to precompile ONSAS [7d8da659-23aa-453a-9a19-a3d1f6c795f5] to "/home/runner/.julia/compiled/v1.11/ONSAS/jl_jfPIrk".
ERROR: LoadError: UndefVarError: `AbstractNode` not defined in `ONSAS.Entities`
```

File: src/Entities/Nodes.jl [hunk 2]
```diff
@@ -89,28 +89,15 @@ function Base.show(io::IO, ::MIME"text/plain", n::Node)
...
-end
-
 "Node constructor with a `NTuple`."
 function Node(t::NTuple{dim,T},
               dofs::Dictionary=Dictionary{Field,Vector{Dof}}()) where {dim,T<:Real}
     Node(Point(t), dofs)
 end
 
+"`Node` constructor to promote types of coordinates of Point"
+Node(t) = Node(promote(t...))
```
Comment (mvanzulli) [inline src/Entities/Nodes.jl]: Minor: Nice, I would suggest to pass also `dofs` as optional here WDYT ? 
```suggestion
Node(t, dofs::Dictionary=Dictionary{Field,Vector{Dof}}()) = Node(promote(t...), dofs)
```

**Review by mvanzulli — COMMENTED** [review]
Thanks for your MR, a couple of minor comments but looks sensible to me.

**Comment by mvanzulli** [comment]
Also, can we create an issue for this PR https://github.com/ONSAS/ONSAS.jl/issues/new/choose ?

File: src/Entities/Entities.jl (hunk 1 above)
Comment (mvanzulli) [inline src/Entities/Entities.jl]:
```suggestion
using ..Nodes
```

**Comment by mvanzulli** [comment]
The current status of CI must succeed to be merged, I would apply the suggestions above.   I'm happy to have meeting if u need help.

File: src/StructuralSolvers/StructuralSolvers.jl [hunk 3]
```diff
@@ -216,7 +216,7 @@ By default `DEFAULT_LINEAR_SOLVER` is utilized.
 """
 function solve!(problem::AbstractStructuralAnalysis,
                 solver::Union{AbstractSolver,Nothing}=nothing,
-                linear_solve::LinearSolver=DEFAULT_LINEAR_SOLVER();
+                linear_solve::LinearSolver=DEFAULT_LINEAR_SOLVER;
```
Comment (mvanzulli) [inline src/StructuralSolvers/StructuralSolvers.jl]: Here the solver is a type, probably that is why we get a method error in CI since is expecting an already instantiated solver object: `DEFAULT_LINEAR_SOLVER()` not the type. I suggest you to go through https://docs.julialang.org/en/v1/manual/methods/ for a full comprehension:  
```suggestion
                linear_solve::LinearSolver=DEFAULT_LINEAR_SOLVER();
```

File: src/Entities/Nodes.jl [hunk 4]
```diff
@@ -89,28 +89,15 @@ function Base.show(io::IO, ::MIME"text/plain", n::Node)
...
-end
-
 "Node constructor with a `NTuple`."
 function Node(t::NTuple{dim,T},
               dofs::Dictionary=Dictionary{Field,Vector{Dof}}()) where {dim,T<:Real}
     Node(Point(t), dofs)
 end
 
+"`Node` constructor to promote types of coordinates of Point"
+Node(t, dofs::Dictionary=Dictionary{Field,Vector{Dof}}()) = Node(promote(t...), dofs)
```
Comment (mvanzulli) [inline src/Entities/Nodes.jl]: Cool so we fixed the previous, now seems is failing because in `t` we have also the `Dictionary` and Julia is using the more general method instead of the internal consttructore. Ref: https://discourse.julialang.org/t/understanding-multiple-dispatch/76601. So I suggest specializing to:
```suggestion
Node(t::Union{Tuple, AbstractArray}, dofs::Dictionary=Dictionary{Field, Vector{Dof}}()) = Node(promote(t...), dofs)
```

**Comment by mvanzulli** [comment]
My email is mcvanzulli@gmail.com, feel free to send me an email so we can have chat 👍🏼

File: src/Entities/Nodes.jl [hunk 5]
```diff
@@ -89,28 +89,15 @@ function Base.show(io::IO, ::MIME"text/plain", n::Node)
...
-end
-
 "Node constructor with a `NTuple`."
 function Node(t::NTuple{dim,T},
               dofs::Dictionary=Dictionary{Field,Vector{Dof}}()) where {dim,T<:Real}
     Node(Point(t), dofs)
 end
 
+"`Node` constructor to promote types of coordinates of Point"
+Node(t::Union{Tuple, AbstractArray}, dofs::Dictionary=Dictionary{Field, Vector{Dof}}()) = Node(promote(t...), dofs)
```
Comment (mvanzulli) [inline src/Entities/Nodes.jl]: The problem is that `t` or usually `args`  type is just a plain: 

```julia
typeof(t) = Tuple{Int64, Int64, Int64, Dictionary{Symbol, Vector{Int64}}}
```
so one can't do a dispatch getter like we tried. Given that, I think what we want is: 
```suggestion
function Node(t)
    dofs = if length(t) > 1 && t[end] isa Dictionary
        t[end]
    else
        Dictionary{Field,Vector{Dof}}()
    end
    coords = t[1:(end - 1)]
    Node(promote(coords)..., dofs)
end

```

**Review by mvanzulli — CHANGES_REQUESTED** [review]

File: src/Entities/Nodes.jl (hunk 5 above)
Comment (mvanzulli) [reply src/Entities/Nodes.jl]: Also I suggest you to apply my changes directly clicking on "commit suggestion" which is most strigthiforwd way to do it.  Please take a look at https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request

File: src/Entities/Nodes.jl [hunk 6]
```diff
@@ -89,28 +89,24 @@ function Base.show(io::IO, ::MIME"text/plain", n::Node)
...
 end
 
+"`Node` constructor to promote types of coordinates of Point"
+function Node(t)
+    dofs = if length(t) > 1 && t[end] isa Dictionary
+        t[end]
+    else
+        Dictionary{Field,Vector{Dof}}()
+    end
+    coords = t[1:(end - 1)]
```
Comment (mvanzulli) [inline src/Entities/Nodes.jl]:
```suggestion
    coords = length(t) > 1 ? t[1:(end - 1)] : (t)
```
